crates/bevy_transform/src/transform_propagate_system.rs

```diff
@@ -29,8 +29,8 @@ fn propagate_recursive(
             transform_query.get::<Transform>(entity),
             transform_query.get_mut::<GlobalTransform>(entity),
         ) {
-            *global_transform.value_mut() = parent * *transform.value();
-            *global_transform.value()
+            global_transform.value = *parent * transform.compute_matrix();
```

Comment:
This requires recomputing the local space transform for every decedent of a parent recursively, which is unnecessary. The original impl cached it in `LocalToParent`. This also can never work with on-change detection because you are forced to recompute every hierarchy in case a descendant changed. Meaning every run you must recompute local space transforms, and global space transforms for every singe entity in a hierarchy regardless of if they changed or not. Especially in development builds where static objects in a hierarchy haven't yet been backed into a `GlobalTransform`, that sounds like an unacceptable performance overhead to me?

Reply:
You're right, I totally overlooked that. Thanks for pointing that out!

Reply:
Now the matrix is only recomputed when the transform changed. This still recomputes all the `GlobalTransform`s every frame, regardless if any changes occured, but this is how the system worked before and should IMO be for a different PR to fix.